nitro-protocol/test/contracts/ForceMove/conclude.test.ts

```diff
@@ -147,26 +151,4 @@ describe('conclude', () => {
       }
     }
   );
-
-  it('Reverts to prevent an underflow', async () => {
```

Comment:
This test refers to [this condition](https://github.com/statechannels/go-nitro/blob/0a7b589b60be963849b52c5560514bd892ab0451/nitro-protocol/contracts/ForceMove.sol#L250), which as I understood, is only used to [prevent underflow](https://github.com/statechannels/go-nitro/blob/0a7b589b60be963849b52c5560514bd892ab0451/nitro-protocol/contracts/ForceMove.sol#L261) when calculating state's turn number.

As we don't construct state hashes inside `ForceMove` anymore, we don't need the check and thus, don't need the test for it.

Reply:
Ok makes sense. But I want to double check: should we have a similar check / test in the shortcutting signature library ? Say in `_requireAcceptableSigsOrder` ?

Reply:
Good point. At first I was sure, there is no chance of uint underflow, but decided to check more thoroughly.
Let's research whether uint overflow is possible in the current logic.

We calculate `shift` as following:
``` solidity
uint256 shift = (largestTurnNum + 1) % nParticipants;
```
and pass it to `_requireAcceptableSigsOrder` function, which guarantees the following inequality is fulfilled:
``` solidity
(signerIndices[i] + nParticipants - shift) % nParticipants <= (turnNum - shift) % nParticipants // (*)
```

We suspect an overflow in parentheses in both sides of inequality.
Let's consider the left one first:
``` solidity
signerIndices[i] + nParticipants - shift
```
An overflow here is not possible, because signerIndices[i] is non-negative and from how `shift` is calculated we can see that `shift < nParticipants` and thus `nParticipants - shift` is always positive.

Now let's take a look at the suspected overflow at the right part of inequality:
``` solidity
turnNum - shift
```
Normally, there should not be any problem calculating this, but some edge case exists.
Suppose we have a channel with 3 participants and provide a support proof (actually invalid), which consists of two states with turn numbers 0 and 1 respectively.
From formula used above, `shift` is equal to 2. We calculate inequality (*) for each state. Thus, calculating turnNum - shift` for the first state results in uint underflow: `uint(0 - 2)`.

To avoid this, we need to add a check for `largestTurnNum + 1` to be bigger or equal than `numParticipants` to `_requireValidInput` function.

Reply:
Thanks for the explanation!